xray/segment.go

```diff
@@ -198,6 +221,33 @@ func (seg *Segment) assignConfiguration(cfg *Config) {
 	seg.Unlock()
 }
 
+// Begin DummySubSegment creates a in the case of no sampling to reduce memory footprint
+func BeginDummySubSegment(ctx context.Context, name string) (context.Context, *Segment) {
+	parent := GetSegment(ctx)
+	if parent == nil {
+		cfg := GetRecorder(ctx)
+		failedMessage := fmt.Sprintf("failed to begin subsegment named '%v': segment cannot be found.", name)
+		if cfg != nil && cfg.ContextMissingStrategy != nil {
+			cfg.ContextMissingStrategy.ContextMissing(failedMessage)
+		} else {
+			globalCfg.ContextMissingStrategy().ContextMissing(failedMessage)
+		}
+		return ctx, nil
+	}
```

Comment:
I think this is necessary because the context passed in dummy subsegment and normal subsegment will not be the same. So this context will return us dummy segment.